include/zephyr/drivers/pcie/pcie.h

```diff
@@ -36,6 +37,23 @@ typedef uint32_t pcie_bdf_t;
  */
 typedef uint32_t pcie_id_t;
 
+struct pcie_dev {
+	pcie_bdf_t bdf;
+	pcie_id_t  id;
+};
+
+#define Z_DEVICE_PCIE_NAME(node_id) _CONCAT(pcie_dev_, DT_DEP_ORD(node_id))
+
+#define PCIE_DT_ID(node_id) PCIE_ID(DT_PROP_OR(node_id, vendor_id, 0xffff), \
```

Comment:
Probably simple comment / API documentation would not hurt.

Reply:
Agreed. I'll add that before promoting to normal PR from draft.